File: Frontend/src/app/DTOs/Create/CreateUpsertSetHeaderDTO.ts
import { SetPolicy } from "src/app/Enums/SetPolicy";
import { Subject } from "src/app/Enums/Subject";

export interface CreateUpsertSetHeaderDTO {
    id: string;
    name: string;
    description: string;
    subjectMain: Subject;
    subjectSecond: Subject | null;
    setPolicy: SetPolicy;
    isEditable: boolean;
}

File: Backend/DTOs/Create/CreateUpsertSetHeaderDTO.cs
namespace Learnz.DTOs;
public class CreateUpsertSetHeaderDTO
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public Subject SubjectMain { get; set; }
    public Subject? SubjectSecond { get; set; }
    public SetPolicy SetPolicy { get; set; }
    public bool IsEditable { get; set; }
    public bool IsPolicyEditable { get; set; }
}